client/state/jetpack-connect/actions.js

```diff
@@ -328,14 +328,13 @@ export function authorize( queryObject ) {
 						site: client_id,
 					} )
 				);
-				debug( 'Sites list updated!', data );
+				debug( 'Site updated', data );
 				dispatch( {
-					type: SITES_RECEIVE,
-					sites: data.sites,
+					type: SITE_RECEIVE,
+					site: data,
 				} );
 				dispatch( {
 					type: JETPACK_CONNECT_AUTHORIZE_RECEIVE_SITE_LIST,
```

Comment:
A nice follow-up will be to drop this action entirely, and just use `SITE_RECEIVE` in the reducer:

https://github.com/Automattic/wp-calypso/blob/d7fde50d0e211e98ff59d6fb1b1b1abadda1da90/client/state/jetpack-connect/reducer/jetpack-connect-authorize.js#L56-L60